samples/bluetooth/broadcaster_multiple/src/main.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/printk.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+/* Maximum supported AD data length, use a value supported by the Controller,
+ * Bluetooth Core Specification define minimum of 31 bytes will be supported by
+ * all Controllers, can be a maximum of 1650 bytes when supported.
+ */
+#if defined(CONFIG_BT_LL_SW_SPLIT)
+#define BT_AD_DATA_LEN_MAX CONFIG_BT_CTLR_ADV_DATA_LEN_MAX
+#else
+#define BT_AD_DATA_LEN_MAX 31
+#endif
+
+/* Device name length, size minus one null character */
+#define BT_DEVICE_NAME_LEN (sizeof(CONFIG_BT_DEVICE_NAME) - 1U)
+
+/* Device name length in AD data format, 2 bytes for length and type overhead */
+#define BT_DEVICE_NAME_AD_DATA_LEN (BT_DEVICE_NAME_LEN + 2U)
+
+/* Maximum manufacturer data length, considering ad data type overhead and
+ * the included device name in ad data format.
+ */
+#define BT_AD_DATA_FORMAT_LEN_MAX 255
+#define BT_MFG_DATA_LEN_MAX       (MIN(BT_AD_DATA_FORMAT_LEN_MAX, \
+				       BT_AD_DATA_LEN_MAX) - 2U)
+#define BT_MFG_DATA_LEN           (MIN(BT_MFG_DATA_LEN_MAX, \
+				       (BT_AD_DATA_LEN_MAX - 2U - \
+					BT_DEVICE_NAME_AD_DATA_LEN)))
+
+static uint8_t mfg_data[BT_MFG_DATA_LEN] = { 0xFF, 0xFF, };
+
+static const struct bt_data ad[] = {
+	BT_DATA(BT_DATA_MANUFACTURER_DATA, mfg_data, sizeof(mfg_data)),
+};
+
+static struct bt_le_ext_adv *adv[CONFIG_BT_EXT_ADV_MAX_ADV_SET];
+
+void main(void)
+{
+	struct bt_le_adv_param adv_param = {
+		.id = BT_ID_DEFAULT,
+		.sid = 0U,
+		.secondary_max_skip = 0U,
+		.options = (BT_LE_ADV_OPT_EXT_ADV | BT_LE_ADV_OPT_USE_NAME),
+		.interval_min = BT_GAP_ADV_FAST_INT_MIN_2,
+		.interval_max = BT_GAP_ADV_FAST_INT_MAX_2,
+		.peer = NULL,
+	};
+	int index;
+	int err;
+
+	printk("Starting Multiple Broadcaster Demo\n");
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	index = CONFIG_BT_EXT_ADV_MAX_ADV_SET;
+	while (index--) {
```

Comment:
For something like this, a `for` loop with `ARRAY_SIZE(adv)` might have been easier to understand (as you are iterating over the `adv` array here).